File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int mxcost=0,minCost=prices[0];
        for(int i=0;i<prices.size();i++)
        {
            minCost=min(minCost,prices[i]);
            mxcost=max(mxcost,prices[i]-minCost);
        }
        return mxcost;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int min_price=prices[0],profit=0;
        for(int i=0;i<prices.length;i++)
        {
            profit=Math.max(profit,prices[i]-min_price);
            min_price=Math.min(min_price,prices[i]);
        }
        return profit;
    }
}